### PR TITLE
kernel: minor cleanups in semaphore and thread 

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
kernel: fix comment section of semaphore object

kernel: fix bit clearing logic in _k_thread_group_leave

### CODE REVIEW

File: kernel/thread.c [hunk 1]
```diff
@@ -577,8 +577,8 @@ void _k_thread_single_abort(struct k_thread *thread)
 	/* Clear initailized state so that this thread object may be re-used
 	 * and triggers errors if API calls are made on it from user threads
 	 */
-	_k_object_uninit(thread);
 	_k_object_uninit(thread->stack_obj);
+	_k_object_uninit(thread);
```
Comment (andrewboie) [inline kernel/thread.c]: why are you changing the order?
these aren't mutexes and all this happens with interrupts locked

Comment (ramakrishnapallala) [reply kernel/thread.c]: i felt it's logical to uninit the object in the dependency order like constructor/destructor. Here in this case we can uninit the thread's stack_obj first and then the thread itself.

Comment (ramakrishnapallala) [reply kernel/thread.c]: andrew, if it does not make sense i can remove the change and resubmit.

**Review by tyrantlizard77 — APPROVED** [review]

File: kernel/thread.c (hunk 1 above)
Comment (andrewboie) [reply kernel/thread.c]: still don't feel like this is quite an appropriate analogy since k_object_uninit isn't a destructor but the change is harmless

**Pull request merged**